### DISCUSSION TITLE
Depending on shared sources

### DISCUSSION THREAD

**Discussion by chrisdonlan** [nvpro-samples/vk_raytracing_tutorial_KHR#26] [Q&A] [answered]
I am working on the tutorial. As I am learning, I am writing new code.  What I am trying to understand is: 

1. How static are the libraries and namespaces available in `shared sources`? (such as `nvvk`, `nvgl`, `nvmath`, etc.)?
2. Can they be safely leveraged as dependencies in the same way I might leverage libraries published in the `nvidia` repo collection, Vulkan, or OpenGL? (in other words, will you guys be supporting and maintaining them?)

Basically, I am trying to understand how deep and close to Vulkan I need to scope any code I write on top of this code that you guys are providing so that I can leverage my GPU.

If it is only good for 1 year or 3 years or whatever, that's cool -- just trying to understand what you guys are going to hold constant and take responsibility for, and where I need to prepare for interfaces to change.

**Comment by mklefrancois** [comment, marked as answer]
Hi @chrisdonlan, I hope the tutorials are helping you and thanks for reporting the issues on Linux, I look at it for the next update (soon).

The code in shared_sources is heavily used and will be maintained for a long time. Many other projects are depending on it and many teams at NVIDIA are using it. We are adapting the code in shared_sources as soon as there are new features, extensions or aware of better techniques.
